Use typeguard during tests and in osc-wrapper.py

dmach opened the pull request.

pep8speaks commented:
Hello @dmach! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:





* In the file [`osc/core.py`](https://github.com/openSUSE/osc/blob/a0028d4dfd6973e638ec99c2ff2a433e13a75180/osc/core.py):

> [Line 3742:64](https://github.com/openSUSE/osc/blob/a0028d4dfd6973e638ec99c2ff2a433e13a75180/osc/core.py#L3742): [E252](https://duckduckgo.com/?q=pep8%20E252) missing whitespace around parameter equals
> [Line 3742:65](https://github.com/openSUSE/osc/blob/a0028d4dfd6973e638ec99c2ff2a433e13a75180/osc/core.py#L3742): [E252](https://duckduckgo.com/?q=pep8%20E252) missing whitespace around parameter equals
> [Line 4199:28](https://github.com/openSUSE/osc/blob/a0028d4dfd6973e638ec99c2ff2a433e13a75180/osc/core.py#L4199): [E252](https://duckduckgo.com/?q=pep8%20E252) missing whitespace around parameter equals
> [Line 4199:29](https://github.com/openSUSE/osc/blob/a0028d4dfd6973e638ec99c2ff2a433e13a75180/osc/core.py#L4199): [E252](https://duckduckgo.com/?q=pep8%20E252) missing whitespace around parameter equals
> [Line 7717:104](https://github.com/openSUSE/osc/blob/a0028d4dfd6973e638ec99c2ff2a433e13a75180/osc/core.py#L7717): [E252](https://duckduckgo.com/?q=pep8%20E252) missing whitespace around parameter equals
> [Line 7717:105](https://github.com/openSUSE/osc/blob/a0028d4dfd6973e638ec99c2ff2a433e13a75180/osc/core.py#L7717): [E252](https://duckduckgo.com/?q=pep8%20E252) missing whitespace around parameter equals

dmach committed:
Enable typeguard during build time and tests, fix issues found

It is also possible to use typeguard in osc-wrapper.py
by setting OSC_TYPEGUARD=1.

dcermak commented on contrib/osc.spec:
```diff
@@ -30,6 +30,13 @@
 %endif
 
+# use typeguard during build on distros where typeguard is available
+%if (0%{?suse_version} > 1500 || 0%{?fedora} >= 37)
```
We don't have to care about Fedora 36 and older
```suggestion
%if (0%{?suse_version} > 1500 || 0%{?fedora})
```

dmach committed:
behave: Fix installing python3-typing-extensions
